media/js/base/core-datalayer-page-id.js

```diff
@@ -22,12 +23,37 @@ if (typeof Mozilla == 'undefined') {
         return pageId ? pageId : pathName.replace(/^(\/\w{2}\-\w{2}\/|\/\w{2,3}\/)/, '/');
     };
 
+    Analytics.getTrafficCopReferrer = function() {
+        // if referrer hasn't been set yet, try to grab it and then remove the cookie
+        if (Mozilla.Cookies && Mozilla.Cookies.hasItem('mozilla-traffic-cop-original-referrer')) {
+            trafficCopOriginalReferrer = Mozilla.Cookies.getItem('mozilla-traffic-cop-original-referrer');
+
+            // referrer shouldn't persist
+            Mozilla.Cookies.removeItem('mozilla-traffic-cop-original-referrer');
+        }
+
+        return trafficCopOriginalReferrer;
```

Comment:
Hmm, this works but I found it a bit confusing when first reading the code. Calling `getTrafficCopReferrer()` twice (line 45 and 48 below) would imply that the cookie gets deleted the first time, however the original referrer is maintained because of the way the variable is scoped. Would this be easier to follow if we avoided side-effects, and let this function return the same thing each time?